flink-runtime/src/main/java/org/apache/flink/runtime/failure/FailureEnricherUtils.java

```diff
@@ -105,6 +105,16 @@ static Collection<FailureEnricher> getFailureEnrichers(
                         includedEnrichers);
             }
         }
+        includedEnrichers.removeAll(
```

Comment:
We could remove the found enricher after line 91.

And shall we change the name of `includedEnrichers` to `enrichersToLoad`. From the perspective of naming, the `includedEnrichers` is meaning the enrichers that user configured, and it shouldn't be modified. But the `enrichersToLoad` means the enrichers that should be load, when some of it is loaded, it could be removed from this set since it's not need load any more.